src/Sylius/Bundle/ResourceBundle/Form/EventSubscriber/AddCodeFormSubscriber.php

```diff
@@ -38,14 +39,20 @@ public static function getSubscribedEvents()
     public function preSetData(FormEvent $event)
     {
         $resource = $event->getData();
+        $type = 'text';
 
         if (!$resource instanceof CodeAwareInterface) {
             throw new UnexpectedTypeException($resource, CodeAwareInterface::class);
         }
 
+        if($resource instanceof CurrencyInterface){
```

Comment:
We definitely can't do it this way. That couples Resource bundle to Currency component. We should be able to pass the ``$type`` of the code field to the subscriber. Just pass it via constructor and say that we use ``text`` as default.